otelconf/internal/provider/envprovider_test.go

```diff
@@ -0,0 +1,231 @@
+// Copyright The OpenTelemetry Authors
+// SPDX-License-Identifier: Apache-2.0
+
+package provider // import "go.opentelemetry.io/contrib/internal/provider"
+
+import (
+	"errors"
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+)
+
+func TestInvalidEnvVarName(t *testing.T) {
+	_, err := replaceEnvVar("$%&(*&)")
+	require.ErrorContains(t, err, errors.New("invalid environment variable name: $%&(*&)").Error())
+}
+
+func TestCheckRawConfTypeNil(t *testing.T) {
+	err := checkRawConfType([]byte{})
+	require.Error(t, err)
+	require.ErrorContains(t, err, "unsupported type=<nil> for retrieved config")
+}
+
+func TestReplaceEnvVar(t *testing.T) {
+	tests := []struct {
+		name     string
+		input    string
+		setupEnv func(t *testing.T)
```

Comment:
nit: all your setups are just setting environment variables. This can be replaced with a `map` of environment variable values that is iterated in the test run.